### PR TITLE
Updated timer to support continuous run (#1)

### PR DESCRIPTION
**Pull request by mohitYogi**
* Updated intervalHandler to support continuous run

intervalHandler is updated in order to support continuous timer run, even after duration is completed

* Flag for single call to callback

Flag for single call to callback when timer completes in continuous run.

* Update utils.js

* support continuous Timer run

1. Added stopTimerOnDuration in config to run timer continuously.
2. Updated intervalHandler to support continuous run
3. Added Flag isCallbackCalled to stop multiple calls to callback after reaching the duration in continuous run.

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -52,7 +52,8 @@ function getDefaultConfig() {
 		repeat: false,				// This will repeat callback every n times duration is elapsed
 		countdown: false,			// If true, this will render the timer as a countdown (must have duration)
 		format: null,				// This sets the format in which the time will be printed
-		updateFrequency: 500		// How often should timer display update
+		updateFrequency: 500,		// How often should timer display update
+            	stopTimerOnDuration: true,	// If set to false than timer will keep on running even after duration is completed.
```
Comment (walmik) [inline src/utils.js]: Is this prop used anywhere? By the way, we do have a prop called `repeat` to do this. Hence we might not need this.

File: src/utils.js [hunk 2]
```diff
@@ -227,11 +228,15 @@ function intervalHandler(timerInstance) {
 
 	if (timerInstance.config.countdown) {
 		timerInstance.totalSeconds = timerInstance.config.duration - timerInstance.totalSeconds;
-
-		if (timerInstance.totalSeconds === 0) {
+		
+		//Added isCallbackCalled flag, to avoid multiple calls to the callback.
+		if (!timerInstance.isCallbackCalled && timerInstance.totalSeconds === 0) {
```
Comment (walmik) [inline src/utils.js]: Do we really need the check for `isCallbackCalled`? We are clearing the interval on the next line. Once the interval is cleared the `callback` should not be called. Here s an example usage of the countdown with duration https://jsbin.com/noyijuqice/edit?js,console,output Lemme know if I m missing something.

File: src/utils.js [hunk 3]
```diff
@@ -240,17 +245,23 @@ function intervalHandler(timerInstance) {
...
 	}
 
 	// If the timer was called with a duration parameter,
 	// run the callback if duration is complete
 	// and remove the duration if `repeat` is not requested
+	
+	// Added new Condition of checking the timer completed or not
 	if (timerInstance.totalSeconds > 0 &&
-		timerInstance.totalSeconds % timerInstance.config.duration === 0) {
+		(timerInstance.totalSeconds % timerInstance.config.duration === 0 || timerInstance.totalSeconds > timerInstance.config.duration )) {
```
Comment (walmik) [inline src/utils.js]: This is a good check to add :)

File: src/utils.js (hunk 1 above)
Comment (mohitYogi) [reply src/utils.js]: removed the prop stopTimerOnDuration.

File: src/utils.js (hunk 2 above)
Comment (mohitYogi) [reply src/utils.js]: Removed the flag.

**Comment by mohitYogi** [comment]
Updated the code.

File: dist/timer.jquery.js [hunk 4]
```diff
@@ -241,7 +241,7 @@ function intervalHandler(timerInstance) {
 		if (timerInstance.totalSeconds === 0) {
 			clearInterval(timerInstance.intervalId);
 			setState(timerInstance, Constants.TIMER_STOPPED);
-			timerInstance.config.callback();
+			timerInstance.config.callback()
```
Comment (walmik) [inline dist/timer.jquery.js]: Looks like the semi-colon got removed here in the dist file.

File: src/utils.js [hunk 5]
```diff
@@ -227,7 +227,8 @@ function intervalHandler(timerInstance) {
 
 	if (timerInstance.config.countdown) {
 		timerInstance.totalSeconds = timerInstance.config.duration - timerInstance.totalSeconds;
-
+		
+		//Added isCallbackCalled flag, to avoid multiple calls to the callback.
```
Comment (walmik) [inline src/utils.js]: We can remove this comment as well.

File: src/utils.js [hunk 6]
```diff
@@ -240,15 +241,18 @@ function intervalHandler(timerInstance) {
...
+	
 	if (!timerInstance.config.duration) {
 		return;
 	}
 
 	// If the timer was called with a duration parameter,
 	// run the callback if duration is complete
 	// and remove the duration if `repeat` is not requested
+	
+	// Added new Condition of checking the timer completed or not
```
Comment (walmik) [inline src/utils.js]: Remove this new comment and update the comment on line 250. Change from,
`// run the callback if duration is complete`
to,
`// run the callback if duration is complete or total seconds is more than duration`

**Comment by mohitYogi** [comment]
Updated files as per required.

**Pull request merged**